math/UsefulMathFunctions.cpp

```diff
@@ -146,43 +166,92 @@ void generatePrimes(int n) {
 // primeDivs[i] will contains all the prime divisors of number i.
 // O(n.log(log(n)))
 vector<int> primeDivs[N];
```

Comment:
Can you write the function (below) such that it return this container? Same thing for your other functions (above). 
Use the dynamic memory managment. ```new``` Operator. Think on deallocating with ```delete```